src/codecs/jpeg/decoder.rs

```diff
@@ -88,7 +85,12 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for JpegDecoder<R> {
     }
 
     fn icc_profile(&mut self) -> Option<Vec<u8>> {
-        self.icc.clone()
+        let mut decoder = zune_jpeg::JpegDecoder::new(&self.input);
+        if let Ok(_) = decoder.decode_headers() {
+            decoder.icc_profile()
+        } else {
+            None
+        }
```

Comment:
This could be:
```suggestion
        decoder.decode_headers().ok()?;
        decoder.icc_profile()
```
But also the method should really return an `ImageResult<Option<Vec<u8>>>` (which unfortunately would be a breaking change, so it'll have to wait for the next major release)

Reply:
I'm not sure the succinct form will work with image's MSRV, let's say what CI says.

It actually cannot fail because the same input was already parsed successfully once. I could have just written an `.unwrap()` there, really.

Reply:
Re: MSRV; the policy is currently to use the field in `Cargo.toml` so it could be bumped. In the spirit of testing ossificiation, maybe we should excercise that with some change beforehand?